bolts-tasks/src/test/java/bolts/TaskTest.java

```diff
@@ -311,6 +312,27 @@ public Integer call() throws Exception {
   }
 
   @Test
+  public void testBackgroundCallWaitingWithTimeouts() throws Exception {
+    final Object sync = new Object();
+
+    Task<Integer> task = Task.callInBackground(new Callable<Integer>() {
+      public Integer call() throws Exception {
+        synchronized (sync) {
+          sync.wait();
```

Comment:
We could do that by adding a short `Thead.sleep(50)` here (ugly, I know :disappointed:) and add another `assertTrue(task.wait...)` after the last one to retain the verification that it returns `true` on an already completed task.